### PR TITLE
Call outputs interim work.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: src/main/scala/cromwell/logging/CromwellLogger.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@ class TerminalLayout extends LayoutBase[ILoggingEvent] {
 
     val highlightedMessage = event.getFormattedMessage
       .replaceAll("UUID\\((.*?)\\)", TerminalUtil.highlight(2, "$1"))
-      .replaceAll("`(.*?)`", TerminalUtil.highlight(9, "$1"))
+      .replaceAll("`(.*?)`", TerminalUtil.highlight(99, "$1"))
```
Comment (mcovarr) [inline src/main/scala/cromwell/logging/CromwellLogger.scala]: Stowaway change, this makes stuff between backticks purple instead of orange that looks too much like red on some displays.

File: src/test/scala/cromwell/engine/WorkflowManagerActorSpec.scala [hunk 2]
```diff
@@ -70,7 +70,7 @@ class WorkflowManagerActorSpec extends CromwellTestkitSpec("WorkflowManagerActor
             val status = if (workflowState == WorkflowSubmitted) NotStarted else Running
             val workflowInfo = new WorkflowInfo(workflowId, wdlSource, wdlInputs)
             val task = new Task("taskName", Seq.empty[Declaration], new Command(Seq.empty), Seq.empty, null, BackendType.LOCAL) // FIXME? null AST
-            val call = new Call(None, key.scope, task, Map.empty)
+          val call = new Call(None, key.scope, task, Map.empty)
```
Comment (mcovarr) [inline src/test/scala/cromwell/engine/WorkflowManagerActorSpec.scala]: will fix

File: src/test/scala/cromwell/engine/db/slick/SlickDataAccessSpec.scala [hunk 3]
```diff
@@ -3,6 +3,7 @@ package cromwell.engine.db.slick
 import java.sql.SQLException
 import java.util.UUID
 
+import classycle.util.NotStringPattern
```
Comment (mcovarr) [inline src/test/scala/cromwell/engine/db/slick/SlickDataAccessSpec.scala]: will fix

File: README.md [hunk 4]
```diff
@@ -714,6 +715,37 @@ Server: spray-can/1.3.3
 }
 ```
 
+
+## GET /workflow/:id/outputs/:call
```
Comment (mcovarr) [inline README.md]: gah should be workflow***s***

File: README.md [hunk 5]
```diff
@@ -714,6 +715,37 @@ Server: spray-can/1.3.3
...
+
+cURL:
+
+```
+$ curl http://localhost:8000/workflow/e442e52a-9de1-47f0-8b4f-e6e565008cf1/outputs/three_step.wc
```
Comment (mcovarr) [inline README.md]: URLs are wrong

File: src/main/scala/cromwell/logging/CromwellLogger.scala (hunk 1 above)
Comment (scottfrazer) [inline src/main/scala/cromwell/logging/CromwellLogger.scala]: The number is fairly arbitrary anyway because terminals can set their own themes so there's very little guarantee about how colors will show up.  I think most implementations honor the first 8 pretty closely, though.  That's where most programs get the yellow for warnings and the red for errors.  Maybe we can choose one of the first 8 color codes, which looks like this on my machine:

screen shot 2015-08-03 at 6 56 57 am

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 6]
```diff
@@ -75,8 +78,41 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
...
+      outputs <- dataAccess.getOutputs(id)
+    } yield {
+      SymbolStoreEntry.toWorkflowOutputs(outputs)
+    }
+  }
+
+  private def callOutputs(workflowId: WorkflowId, callFqn: String): Future[binding.CallOutputs] = {
+    for {
+      _ <- assertWorkflowExistence(workflowId)
+      _ <- assertCallExistence(workflowId, callFqn)
```
Comment (scottfrazer) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: I'm not sure I understand why the assertions are here.

Comment (scottfrazer) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: Oh I think I see... because the user might specify a bogus workflowId or callFqn?

Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: yup

File: src/main/scala/cromwell/logging/CromwellLogger.scala (hunk 1 above)
Comment (mcovarr) [inline src/main/scala/cromwell/logging/CromwellLogger.scala]: Per discussion with Scott we're going to try `ansi=5` above.

**Comment by mcovarr** [comment]
Suggested fixes made, should be ready for a second reviewer.

File: src/main/scala/cromwell/binding/package.scala [hunk 7]
```diff
@@ -21,8 +21,9 @@ package object binding {
   type WorkflowCoercedInputs = Map[FullyQualifiedName, WdlValue]
   type WorkflowOutputs = Map[FullyQualifiedName, WdlValue]
   type FullyQualifiedName = String
+  type LocallyQualifiedName = String
```
Comment (geoffjentry) [inline src/main/scala/cromwell/binding/package.scala]: Since we've both grumbled about the type aliases, would now be a good time to stop using them in favor of things like case class wrappers?

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala [hunk 8]
```diff
@@ -394,13 +407,13 @@ class SlickDataAccess(databaseConfig: Config, val dataAccess: DataAccessComponen
...
-    getSymbols(workflowId, IoInput, Option(call))
+    getSymbols(workflowId, IoInput, Option(call.fullyQualifiedName))
   }
 
   /** Get all outputs for the scope of this call. */
-  override def getOutputs(workflowId: WorkflowId, call: Call): Future[Traversable[SymbolStoreEntry]] = {
-    require(call != null, "call cannot be null")
-    getSymbols(workflowId, IoOutput, Option(call))
+  override def getOutputs(workflowId: WorkflowId, callFqn: FullyQualifiedName): Future[Traversable[SymbolStoreEntry]] = {
+    require(callFqn != null, "callFqn cannot be null")
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Why would it be null?

File: src/main/scala/cromwell/webservice/CromwellApiService.scala [hunk 9]
```diff
@@ -203,4 +203,36 @@ trait CromwellApiService extends HttpService with PerRequestCreator {
...
       }
     }
+
+  @Path("/{version}/{workflowId}/outputs/{callFqn}")
+  @ApiOperation(
+    value = "Query for call outputs based on workflow id and call fully qualified name (e.g. my_workflow.my_call).",
+    nickname = "call-outputs",
+    httpMethod = "GET",
+    produces = "application/json"
+    //,response = classOf[CallOutputs]
```
Comment (geoffjentry) [inline src/main/scala/cromwell/webservice/CromwellApiService.scala]: Why is this here?

Comment (scottfrazer) [inline src/main/scala/cromwell/webservice/CromwellApiService.scala]: I believe this was commented out of every annotation like this a while ago because we couldn't get something working (Swagger related?) but other uses of this annotation in other projects used the `response =`, so we left it in as a reminder or something... Probably just cruft at this point though.

Comment (geoffjentry) [inline src/main/scala/cromwell/webservice/CromwellApiService.scala]: I really meant "why is this still here" :)

File: src/main/scala/cromwell/binding/package.scala (hunk 7 above)
Comment (mcovarr) [inline src/main/scala/cromwell/binding/package.scala]: I agree with the move to case classes, though perhaps that could be a ticket to replace them consistently.

File: src/main/scala/cromwell/webservice/CromwellApiService.scala (hunk 9 above)
Comment (mcovarr) [inline src/main/scala/cromwell/webservice/CromwellApiService.scala]: I assumed this was a reminder to uncomment this once we had a Swagger/Akka HTTP stack that could handle its presence (see @kshakir's findings in the class Scaladoc at the top of this file).

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala (hunk 8 above)
Comment (mcovarr) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: I'm not sure how it could be null in real life but let's loop in the author (@kshakir).

File: src/main/scala/cromwell/binding/package.scala (hunk 7 above)
Comment (mcovarr) [inline src/main/scala/cromwell/binding/package.scala]: Created DSDEEPB-956

Comment (geoffjentry) [inline src/main/scala/cromwell/binding/package.scala]: Agreed in terms of removing old ones but why create more debt here? There's no harm in doing it the future-correct-way here.

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala (hunk 8 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: If the answer is "it might come in as null from outside the system" let's make these checks happen on ingest

File: src/main/scala/cromwell/binding/package.scala (hunk 7 above)
Comment (mcovarr) [inline src/main/scala/cromwell/binding/package.scala]: Yeah, true.  I'll try to slip this in to my DSDEEPB-29 work.

File: src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala (hunk 8 above)
Comment (kshakir) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Without this check, the behavior of an accidental `call = null` would return _ALL_ workflow outputs. I didn't want that to happen even accidentally, because I thought it would be a pain to debug. So I threw in the `require()`, instead throwing a `IllegalArgumentException`.

Comment (geoffjentry) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: It'd be good to avoid `null` as much as possible even if to stay out of the habit. I'll note that `Option(null)` is automatically a `None`, perhaps we could do something w/ that here.

Comment (kshakir) [inline src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala]: Under the current code flow, as long as that `None` doesn't make it to `getSymbols`, I'm fine with whatever refactoring.
